### PR TITLE
Milestone1

### PR DESCRIPTION
**Pull request by AtillaTahak**
**HELLO REVİEWER**

Create class Person with the following:
Instance vars: @id, @name, and @age.
Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
Getters for @id, @name, and @age.
Setters for @name and @age.
Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
Public method can_use_services? that returns true if person is of age or if they have permission from parents.
Create class Student with the following:
Inherits from Person.
Constructor extends parent's constructor by adding @classroom and a parameter for it.
Method play_hooky that returns "¯\(ツ)/¯".
Create class Teacher with the following:
Inherits from Person.
Constructor extends parent's constructor by adding @specialization and a parameter for it.
Override can_use_services? so it always returns true.
Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name: 'Unknown', parent_permission: 'Unknown')
+    @id = Ramdom.rand(1..1000)
```
Comment (andersonlebon) [inline person.rb]: - [ ] Kindly note that the instance variable `@parent_permission` should be a boolean and should be true by default, not `unknown`.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    @parent_permission == 'allowed'
+  end
```
Comment (andersonlebon) [inline person.rb]: - [ ] One of the requirements says that the `can_use_services?` method should be public the only private method for the Person class is `of_age?`. Kindly remove the `can_use_services?` method in the private.

Comment (andersonlebon) [inline person.rb]: - [ ] Kindly note that the logic behind this method `can_use_services?` is that it should return true if the @age >= 18  or if they have permission from parents(`@parent_permission == true`).
```rb
def can_use_services? 
    of_age? || @parent_permission
  end
```

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(classroom)
+    super
+    @classroom = classroom
+  end
```
Comment (andersonlebon) [inline student.rb]: - [ ] Kindly note that as this class enherits from Persone class its constructor should extend all parent's @instance variables comment in the children initialize parameter. :point_down: 

```rb
class Student < Person
   def initialize(age, classroom, name = "Unknown", parent_permission = true)  
        super(age, name, parent_permission)
          @classroom = classroom
    end
end
```

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(specialization)
+    super
+    @specialization = specialization
+  end
```
Comment (andersonlebon) [inline teacher.rb]: #### The same here :point_down: 

- [ ] Kindly note that as this class enherits from Persone class its constructor should extend all parent's @instance variables comment in the children initialize parameter. :point_down: 

```rb
class Teacher < Person
    def initialize(age, specialization, name = "Unknown", parent_permission = true)
       super(age, name, parent_permission)
       @specialization = specialization
     end
end
```

File: person.rb (hunk 1 above)
Comment (andersonlebon) [inline person.rb]: - [ ] Kindly note that the instance variable `@parent_permission` should be a boolean and should be true by default, not `unknown`.

File: README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ![](https://img.shields.io/badge/Microverse-blueviolet)
 
-# Project Name
+# School Library
```
Comment (andersonlebon) [inline README.md:3]: - [ ] Kindly provide information for empty sections in your readme file to make it more professional.

image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @AtillaTahak 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (rloterh) [reply person.rb]: This issue is well resolved! ✔️

File: person.rb (hunk 2 above)
Comment (rloterh) [reply person.rb]: This issue is well resolved! ✔️

Comment (rloterh) [reply person.rb]: This issue is well resolved! ✔️

File: student.rb (hunk 3 above)
Comment (rloterh) [reply student.rb]: This issue is well resolved! ✔️

File: teacher.rb (hunk 4 above)
Comment (rloterh) [reply teacher.rb]: This issue is well resolved! ✔️

File: person.rb (hunk 1 above)
Comment (rloterh) [reply person.rb]: This issue is well resolved! ✔️

File: README.md (hunk 5 above)
Comment (rloterh) [reply README.md:3]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Atilla, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**